analyzer_test.go

```diff
@@ -27,551 +27,654 @@ import (
 )
 
 func TestAnalyzer(t *testing.T) {
-	spec.Run(t, "Analyzer", testAnalyzer, spec.Report(report.Terminal{}))
+	for _, api := range api.Platform.Supported {
+		spec.Run(t, "unit-analyzer/"+api.String(), testAnalyzerBuilder(api.String()), spec.Parallel(), spec.Report(report.Terminal{}))
+	}
 }
 
-func testAnalyzer(t *testing.T, when spec.G, it spec.S) {
-	var (
-		analyzer  *lifecycle.Analyzer
-		mockCtrl  *gomock.Controller
-		layerDir  string
-		tmpDir    string
-		cacheDir  string
-		testCache lifecycle.Cache
-	)
-
-	it.Before(func() {
-		var err error
-
-		tmpDir, err = ioutil.TempDir("", "analyzer-tests")
-		h.AssertNil(t, err)
-
-		layerDir, err = ioutil.TempDir("", "lifecycle-layer-dir")
-		h.AssertNil(t, err)
-
-		cacheDir, err = ioutil.TempDir("", "some-cache-dir")
-		h.AssertNil(t, err)
-
-		testCache, err = cache.NewVolumeCache(cacheDir)
-		h.AssertNil(t, err)
-
-		analyzer = &lifecycle.Analyzer{
-			Buildpacks: []buildpack.GroupBuildpack{
-				{ID: "metadata.buildpack", API: api.Buildpack.Latest().String()},
-				{ID: "no.cache.buildpack", API: api.Buildpack.Latest().String()},
-				{ID: "no.metadata.buildpack", API: api.Buildpack.Latest().String()},
-			},
-			LayersDir: layerDir,
-			Logger:    &log.Logger{Handler: &discard.Handler{}},
-		}
-		if testing.Verbose() {
-			analyzer.Logger = cmd.DefaultLogger
-			h.AssertNil(t, cmd.SetLogLevel("debug"))
-		}
-		mockCtrl = gomock.NewController(t)
-	})
-
-	it.After(func() {
-		h.AssertNil(t, os.RemoveAll(tmpDir))
-		h.AssertNil(t, os.RemoveAll(layerDir))
-		h.AssertNil(t, os.RemoveAll(cacheDir))
-		mockCtrl.Finish()
-	})
-
-	when("#Analyze", func() {
+func testAnalyzerBuilder(apiString string) func(t *testing.T, when spec.G, it spec.S) {
```

Comment:
```suggestion
func testAnalyzerBuilder(platformApiString string) func(t *testing.T, when spec.G, it spec.S) {
```

Reply:
Maybe the "string" part is unnecessary and it's better to rename it to `platformAPI`.